src/plugin/helpers/validation.ts

```diff
@@ -19,46 +20,72 @@ export const createValidator = (
   return errors;
 };
 
-export const htmlMaxLength = (maxLength: number): Validator => (value) => {
+export const htmlMaxLength = (
+  maxLength: number,
+  customMessage: string | undefined = undefined
+): Validator => (value) => {
   if (typeof value !== "string") {
     throw new Error(`[htmlMaxLength]: value is not of type string`);
   }
   const el = document.createElement("div");
   el.innerHTML = value;
   if (el.innerText.length > maxLength) {
-    return [`Too long: ${el.innerText.length}/${maxLength}`];
+    return [
+      {
+        error: `Too long: ${value.length}/${maxLength}`,
+        message:
+          customMessage ?? `Property is too long: ${value.length}/${maxLength}`,
+      },
+    ];
   }
   return [];
 };
 
-export const maxLength = (maxLength: number): Validator => (value) => {
+export const maxLength = (
+  maxLength: number,
+  customMessage: string | undefined = undefined
+): Validator => (value) => {
   if (typeof value !== "string") {
     throw new Error(`[maxLength]: value is not of type string`);
   }
   if (value.length > maxLength) {
-    return [`Too long: ${value.length}/${maxLength}`];
+    return [
+      {
+        error: `Too long: ${value.length}/${maxLength}`,
+        message:
+          customMessage ?? `Property is too long: ${value.length}/${maxLength}`,
```

Comment:
Would it be worth us passing the field name in when validating, so we could  write `${fieldName} is too long` and make the default more usable?

Reply:
Haha I had exactly that and took it out. I think it's a good idea, happy to add it back.

Reply:
@jonathonherbert I've added this back in. I think I removed it as the field name may be both internal and none human friendly i.e. camel case, but I see little harm in having it as a default. Let me know what you think.